src/components/CategorySelector.tsx

```diff
@@ -93,19 +98,45 @@ export const CategorySelector = ({
           InputProps={{ endAdornment: <SearchIcon /> }}
           onChange={(event) => setSearchValue(event.target.value)}
         />
-        <Box
-          sx={{
-            display: 'flex',
-            gap: '0.5rem',
-            alignItems: 'center',
-          }}>
-          <FilterVintageIcon
+        {setSelectedCategories ? (
+          <Chip
+            icon={
+              <FilterVintageIcon
+                color="primary"
+                titleAccess={t('registration.resource_type.nvi.can_give_publication_points')}
+                fontSize="small"
+              />
+            }
             color="primary"
-            titleAccess={t('registration.resource_type.nvi.can_give_publication_points')}
-            fontSize="small"
+            title={t('registration.resource_type.nvi.select_all_nvi_categories')}
+            onClick={() => {
+              if (!highlightNviCategories && setSelectedCategories) {
+                setSelectedCategories([...selectedCategories, ...nviApplicableTypes]);
+              } else if (setSelectedCategories) {
```

Comment:
Nå har vi denne sjekken på linje 101 i stedet, så `setSelectedCategories` vil aldri bli false her
```suggestion
              if (!highlightNviCategories) {
                setSelectedCategories([...selectedCategories, ...nviApplicableTypes]);
              } else {
```